db/blob/blob_log_writer.cc

```diff
@@ -150,28 +173,34 @@ void BlobLogWriter::ConstructBlobHeader(std::string* buf, const Slice& key,
   record.EncodeHeaderTo(buf);
 }
 
-Status BlobLogWriter::EmitPhysicalRecord(const std::string& headerbuf,
+Status BlobLogWriter::EmitPhysicalRecord(const WriteOptions& write_options,
+                                         const std::string& headerbuf,
                                          const Slice& key, const Slice& val,
                                          uint64_t* key_offset,
                                          uint64_t* blob_offset) {
-  StopWatch write_sw(clock_, statistics_, BLOB_DB_BLOB_FILE_WRITE_MICROS);
-  Status s = dest_->Append(Slice(headerbuf));
+  IOOptions opts;
+  Status s = WritableFileWriter::PrepareIOOptions(write_options, opts);
   if (s.ok()) {
-    s = dest_->Append(key);
+    s = dest_->Append(opts, Slice(headerbuf));
   }
   if (s.ok()) {
-    s = dest_->Append(val);
+    s = dest_->Append(opts, key);
+  }
+  if (s.ok()) {
+    s = dest_->Append(opts, val);
   }
   if (do_flush_ && s.ok()) {
-    s = dest_->Flush();
+    s = dest_->Flush(opts);
```

Comment:
Moving BLOB_DB_BLOB_FILE_WRITE_MICROS from above to WritableFileWriter::Append() will not measure this line. This should only affect legacy BlobDB. as do_flush is only used there. See https://github.com/facebook/rocksdb/commit/431e8afba7fc42c55d03a00da00f848560b20de0 for more.